samples/clip-geometry/src/main/java/com/esri/arcgismaps/sample/clipgeometry/components/ClipGeometryViewModel.kt

```diff
@@ -0,0 +1,177 @@
+/* Copyright 2024 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.clipgeometry.components
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.geometry.Envelope
+import com.arcgismaps.geometry.Geometry
+import com.arcgismaps.geometry.GeometryEngine
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.symbology.SimpleFillSymbol
+import com.arcgismaps.mapping.symbology.SimpleFillSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.clipgeometry.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+
+class ClipGeometryViewModel(application: Application) : AndroidViewModel(application) {
+    // create  a map with a topographic basemap style
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISTopographic)
+    // create a MapViewProxy to interact with the MapView
+    val mapViewProxy = MapViewProxy()
+
+
+
+    private var coloradoGraphic  = createColoradoGraphic()
+    private var coloradoFillSymbol = SimpleFillSymbol(
+        SimpleFillSymbolStyle.Solid,
+        Color(R.color.transparentDarkBlue),
+        SimpleLineSymbol(SimpleLineSymbolStyle.Solid, Color.green, 2f)
+    )
+    private var envelopesGraphics = createEnvelopeGraphics()
+
+
+    // graphics overlay to display graphics
+    val graphicsOverlay = GraphicsOverlay().apply {
+        graphics.add(coloradoGraphic)
+        graphics.addAll(envelopesGraphics)
+    }
+
+    // Create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    val isClipButtonEnabled = mutableStateOf(true)
+    val isResetButtonEnabled = mutableStateOf(false)
```

Comment:
Instead of declaring these as `MutableState<Boolean>`, you could use delegation:
```suggestion
    var isClipButtonEnabled by mutableStateOf(true)
    var isResetButtonEnabled by mutableStateOf(false)
```
That way, you can access and set these directly as booleans, rather than needing to call `.value` every time.

You'd just need these imports:
```
import androidx.compose.runtime.getValue
import androidx.compose.runtime.setValue
```

Reply:
It's also a little strange that the clip button is enabled before the map is loaded. You could have that false initially, then set it to true in an `onSuccess` for the map.